quickwit/quickwit-ingest/src/ingest_v2/ingester.rs

```diff
@@ -961,6 +961,48 @@ impl Ingester {
 
         Ok(DecommissionResponse {})
     }
+
+    pub async fn mrecordlog_summary(&mut self) -> IngestV2Result<MRecordlogSummary> {
+        let rw_mrecordlog = self.state.mrecordlog();
+        // this is a debug api endpoint, with_lock_metrics! doesn't seem necessary
+        let maybe_mrecordlog = rw_mrecordlog.read().await;
+
+        let summary = match &*maybe_mrecordlog {
+            Some(mrecordlog) => mrecordlog.summary(),
+            None => {
+                return Err(IngestV2Error::Internal(
+                    "mrecordlog isn't initialized".to_string(),
+                ))
+            }
+        };
+        drop(maybe_mrecordlog);
+
+        let queues = summary
+            .queues
+            .into_iter()
+            .map(|(id, summary)| MRecordlogQueueSummary {
+                id,
+                start: summary.start,
+                end: summary.end,
+                file_number: summary.file_number,
+            })
+            .collect();
+
+        Ok(MRecordlogSummary { queues })
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub struct MRecordlogSummary {
+    pub queues: Vec<MRecordlogQueueSummary>,
```

Comment:
Nitpick. You did not like the format of the QuerySummary in record log? I think we could have modified instead of having 2.

Reply:
for objects inside the language hashmap are better, but for APIs, i tend to prefer lists of key-values. I don't think it really matter, i'm going to use what we have in mrecordlog directly